### PR TITLE
[dotnet] change how we get library version

### PR DESCRIPTION
**Pull request by lucaspimentel**
## Motivation

We had an issue in .NET where we reported the wrong library version (or it was in an invalid format, unclear). I couldn't reproduce this issue, but while we were all looking into it, we noticed we could make several improvements to how we detect and report the library version to prevent this from happening and making everything less brittle.

The main issue is that we currently use the `strings` tool from the `binutils` package and a regular expression to extract the first `a.b.c.d` version string from `Datadog.Trace.dll`. This is not great since a string in that format could be found multiple times all over the file and we have no guarantee that we are using the correct one. Also, we were saving this string into a file and reading it from several places, even though in .NET we can access the assembly version directly.

## Changes

tl;dr
- library version is no longer required at build time or saved to disk
- we use `Assembly.LoadFile()` to get the library version
- `weblog` tests retrieve the version using the `/healthcheck` endpoint
- `parametric` tests uses a custom .NET tool to extract the version

full details
- stop installing the `binutils` package and using the `strings` tool
- stop saving the library version to a file in `install_ddtrace.sh`
- `weblog`
  - always define the build-time constants `DDTRACE_2_7_0_OR_GREATER` and `DDTRACE_2_23_0_OR_GREATER` since everything is 3.x now
  - return the correct version at runtime in the `/healthcheck` endpoint
  - TODO LATER: remove these constants and the code that uses them
- `parametric` scenario
  - runs `system_tests_library_version.sh`, which uses a custom .NET tool to load the assembly and retrieve it's version (Node.js, Python, Go, and Ruby also use a similar script)

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
After the change I can see the library version in the endtoend scenarios:
```Library: dotnet@3.8.0+0```
In the main branch execution I see:
```Library: dotnet@3.8.0```
After the change  I can see the library version in the paramettric scenario:
```Library: dotnet@3.8.0```
is this correct?

File: utils/build/docker/dotnet/weblog/Endpoints/HealthcheckEndpoint.cs [hunk 1]
```diff
@@ -12,17 +11,15 @@ public void Register(Microsoft.AspNetCore.Routing.IEndpointRouteBuilder routeBui
         {
             routeBuilder.MapGet("/healthcheck", async context =>
             {
-                string dd_version = "";
-                using (StreamReader reader = new StreamReader("SYSTEM_TESTS_LIBRARY_VERSION"))
-                {
-                    dd_version = reader.ReadToEnd().Trim( new Char[] { '\n' } );
-                }
+                var version = Assembly.Load("Datadog.Trace").GetName().Version?.ToString(4);
```
[the remark this answers is not in the document]
Comment (lucaspimentel) [reply utils/build/docker/dotnet/weblog/Endpoints/HealthcheckEndpoint.cs:14]: That's it, thanks. I noticed this issue and I thought I had changed it from 4 to 3. Seems like I lost that commit somewhere 😬

[the remark this answers is not in the document]
Comment (lucaspimentel) [reply utils/build/docker/dotnet/weblog/Endpoints/HealthcheckEndpoint.cs:14]: Oh, this in in `weblog`. Thanks!

**Review by cbeauchesne — APPROVED** [review]

File: utils/build/docker/dotnet/weblog/Endpoints/HealthcheckEndpoint.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucaspimentel) [reply utils/build/docker/dotnet/weblog/Endpoints/HealthcheckEndpoint.cs]: @robertomonteromiguel this fixes the issue in your comment.

**Review by robertomonteromiguel — APPROVED** [review]
